File: backend/app/models/search_session.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, JSON
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.db.base import Base

class SearchSession(Base):
    __tablename__ = "search_sessions"

    search_id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.user_id"), nullable=False)

    search_query = Column(String, nullable=False)
    search_location = Column(String, nullable=True)
    search_filters = Column(JSON, nullable=True)

    context_id = Column(Integer, ForeignKey('search_contexts.id'), nullable=True)
    context = relationship("SearchContext", back_populates="sessions")

    google_api_request_hash = Column(String, index=True)
    next_page_token = Column(String, nullable=True)
    result_count = Column(Integer, default=0)

    exporter_profile_id = Column(
        Integer,
        ForeignKey("exporter_profiles.id"),
        nullable=True
    )

    ai_context = Column(String, nullable=True)
    approved_queries = Column(JSONB, nullable=True)
    discovery_platform = Column(String(10), nullable=True, default="both")

    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: front_end/src/types/search-session.ts
export interface SearchSession {
  search_id: number;
  user_id?: number;
  query?: string;
  search_query?: string;
  search_location?: string | null;
  search_filters?: Record<string, unknown> | null;
  context_id?: number | null;
  next_page_token?: string | null;
  total_results?: number | null;
  created_at?: string | null;
}

export const getSearchSessionQuery = (session: SearchSession) =>
  session.query || session.search_query || `Search #${session.search_id}`;
